db/db.go

```diff
@@ -76,623 +70,746 @@ func New(db *sql.DB) DB {
 	if err != nil {
 		return nil, err
 	}
-	defer shiftIns.Close()
+	defer rows.Close()
 
-	// Loop through rows and convert them to txs.
-	for shiftIns.Next() {
-		tx, err := rowToShiftIn(shiftIns)
-		if err != nil {
-			return nil, err
-		}
-		tx, err = db.autogen(tx)
+	for rows.Next() {
+		transaction, err := rowToBurnAccountReleaseUTXOTx(rows)
 		if err != nil {
 			return nil, err
 		}
-		txs = append(txs, tx)
+		txs = append(txs, transaction)
 	}
 
-	return txs, shiftIns.Err()
-}
-
-// UnsubmittedTxs implements the `DB` interface.
-func (db database) UnsubmittedTxs(expiry time.Duration) ([]abi.B32, error) {
-	hashes := make([]abi.B32, 0)
-
-	// Get txs which haven't been submitted
-	shiftIns, err := db.db.Query(`SELECT hash FROM shift_in 
-		WHERE status = $1 AND $2 - created_time < $3 AND gaas=TRUE;`, TxStatusConfirmed, time.Now().Unix(), int64(expiry.Seconds()))
+	// Get pending burn-and-release (account -> account) transactions from the database.
+	rows, err = db.db.Query(`SELECT hash, selector, amount, toAddr, nonce FROM burn_account_release_account
+	WHERE status = $1 AND $2 - created_time < $3`, TxStatusConfirming, time.Now().Unix(), int64(expiry.Seconds()))
 	if err != nil {
 		return nil, err
 	}
-	defer shiftIns.Close()
+	defer rows.Close()
 
-	for shiftIns.Next() {
-		var hash string
-		if err := shiftIns.Scan(&hash); err != nil {
-			return nil, err
-		}
-		txHash, err := stringToB32(hash)
+	for rows.Next() {
+		transaction, err := rowToBurnAccountReleaseAccountTx(rows)
 		if err != nil {
 			return nil, err
 		}
-		hashes = append(hashes, txHash)
+		txs = append(txs, transaction)
 	}
-	return hashes, shiftIns.Err()
+	return txs, rows.Err()
 }
 
-// TxStatus implements the `DB` interface.
-func (db database) TxStatus(hash abi.B32) (TxStatus, error) {
+// TxStatus implements the DB interface.
+func (db database) TxStatus(txHash pack.Bytes32) (TxStatus, error) {
 	var status int
-	err := db.db.QueryRow(`SELECT status FROM shift_in WHERE hash = $1;`,
-		hex.EncodeToString(hash[:])).Scan(&status)
-	if err == sql.ErrNoRows {
-		err = db.db.QueryRow(`SELECT status FROM shift_out WHERE hash = $1;`,
-			hex.EncodeToString(hash[:])).Scan(&status)
+	err := db.db.QueryRow(`SELECT status FROM lock_utxo_mint_account WHERE hash = $1;`, txHash.String()).Scan(&status)
+	if err != sql.ErrNoRows {
+		return TxStatus(status), err
+	}
+	err = db.db.QueryRow(`SELECT status FROM lock_account_mint_account WHERE hash = $1;`, txHash.String()).Scan(&status)
+	if err != sql.ErrNoRows {
+		return TxStatus(status), err
+	}
+	err = db.db.QueryRow(`SELECT status FROM burn_account_release_utxo WHERE hash = $1;`, txHash.String()).Scan(&status)
+	if err != sql.ErrNoRows {
+		return TxStatus(status), err
+	}
+	err = db.db.QueryRow(`SELECT status FROM burn_account_release_account WHERE hash = $1;`, txHash.String()).Scan(&status)
+	if err != sql.ErrNoRows {
+		return TxStatus(status), err
 	}
```

Comment:
Consider using a `JOIN` statement so that only one query ever needs to get executed. Although, off the top of my head, I am not sure which kind of join you need here.